### PR TITLE
Adding AOT Autograd backend for inference

### PR DESCRIPTION
**Pull request by anijain2305**
@chillee @jansel Added AOT Autograd backend. Let me know if you have any alternate suggestions for the API.

I have tested this for inference for now. It might go through some changes as we work on training.

### CODE REVIEW

File: torchdynamo/optimizations/backends.py [hunk 1]
```diff
@@ -452,6 +452,16 @@ def run(*new_inputs):
     return run
 
 
+@create_backend
+def aot_autograd(subgraph, **kwargs):
+    from functorch.compile import aot_function, memory_efficient_fusion
```
Comment (anijain2305) [inline torchdynamo/optimizations/backends.py:457]: I kept the imports inside for now to avoid making functorch a dependency.

**Review by jansel — APPROVED** [review]
LGTM.  Need to land the other PR first though.

File: torchdynamo/optimizations/backends.py (hunk 1 above)
Comment (jansel) [reply torchdynamo/optimizations/backends.py:457]: That's perfect

File: torchdynamo/optimizations/backends.py [hunk 2]
```diff
@@ -452,6 +452,16 @@ def run(*new_inputs):
     return run
 
 
+@create_backend
+def aot_autograd(subgraph, **kwargs):
```
Comment (jansel) [inline torchdynamo/optimizations/backends.py:456]: Note the `@create_backend` decorator might prevent `**kwargs` from working.

Comment (anijain2305) [reply torchdynamo/optimizations/backends.py:456]: It seems to be working for my testcases. But, I will double check.

Comment (jansel) [reply torchdynamo/optimizations/backends.py:456]: Ah cool, I wasn't sure.

**Comment by Chillee** [comment]
Out of curiosity, what makes it not immediately work for training?

**Comment by anijain2305** [comment]
> Out of curiosity, what makes it not immediately work for training?

Actually, it works out of the box here. I am meeting Jason to figure out if there are any gaping holes, or any alternate design.

**Comment by jansel** [comment]
> Actually, it works out of the box here. I am meeting Jason to figure out if there are any gaping holes, or any alternate design.

Oh cool.  I'm actually kind of expecting it to be this easy (these ~5 lines to add the backend).  I feel like 90% of the work will be in measurment and fixing the odd bug or two that surfaces.

**Pull request merged**